jaxrs/WFLY-13530_expose_JAXRS_to_gRPC.adoc

```diff
@@ -0,0 +1,133 @@
+= Expose JAX-RS Resources to gRPC clients
+:author:            Ron Sigal
+:email:             rsigal@redhat.com
+:toc:               left
+:icons:             font
+:idprefix:
+:idseparator:       -
+:issue-base-url:    https://issues.jboss.org/browse
+
+== Overview
+
+In the five years since gRPC was introduced, it has become a popular invocation
+and streaming framework, used by major corporations. Among its benefits are
+
+* speed
+* language independence
+
+With this feature request we propose to expose JAX-RS services as gRPC services,
+accessible by gRPC clients.
+
+Given a gRPC service definition like
```

Comment:
The kind of feature requests we see seem to be asking about gRPC in terms of using gRPC for existing deployments, it that going to trigger a need for generation of these definitions?

Reply:
Another great point. Yeah, to create a gRPC client, there would have to be a protobuf definition (barring something magical, I guess). I was working on a translator that would take a Java class and translate it into a protobuf definition. It works, but only for a very limited set of classes, for now. Also, I was thinking at the protubuf level, which doesn't involve methods. So, for now, I'm thinking that the protobuf definition would be created manually. I think that automating it would be an interesting project, though.